ddtrace/internal/writer/writer.py

```diff
@@ -540,38 +539,32 @@ def agent_url(self):
     def _agent_endpoint(self):
         return self._intake_endpoint(client=None)
 
-    def _downgrade(self, payload, response, client):
+    def _downgrade(self, response, client):
```

Comment:
```suggestion
    # TODO(munir): v0.5 endpoint was introduced in ddtrace v7.22.0 (August 2020). The vast majority of 
    # users should be using a Datadog Agent that supports the v0.5 endpoint. Soon we should deprecate and
    # remove support for automatically downgrading the trace api. Instead we should ask user's to upgrade
    # their datadog agent or prompt them to set DD_TRACE_API_VERSION to v0.4
    def _downgrade(self, response, client):
```

Reply:
This is overkill. If we introduce a new trace api this function could be useful!